format-xls/src/main/java/io/cdap/plugin/format/xls/input/XlsInputFormatProvider.java

```diff
@@ -80,11 +81,15 @@ public void validate(FormatContext context) {
     Schema schema = super.getSchema(context);
     FailureCollector collector = context.getFailureCollector();
     // When the sheet is specified by number, the sheet value must be a number
-    if (!conf.containsMacro(XlsInputFormatConfig.NAME_SHEET_VALUE)
-            && conf.getSheet().equals(XlsInputFormatConfig.SHEET_NUMBER)
-                    && (conf.getSheetValue() == null || !conf.getSheetValue().matches("[0-9]+"))) {
-      collector.addFailure("Sheet number must be a number.", null)
-              .withConfigProperty(XlsInputFormatConfig.NAME_SHEET_VALUE);
+    if (!conf.containsMacro(XlsInputFormatConfig.NAME_SHEET_VALUE) &&
+            conf.getSheet().equals(XlsInputFormatConfig.SHEET_NUMBER) &&
+            !Strings.isNullOrEmpty(conf.getSheetValue())) {
+      try {
+        Integer.parseInt(conf.getSheetValue());
```

Comment:
while we're doing this, can also check that it's not negative

Reply:
```java
 try {
        int sheetValue = Integer.parseInt(conf.getSheetValue());
        if (sheetValue >= 0) {
          return sheetValue;
        }
        failureCollector.addFailure("Sheet number must be a positive number.", null)
                    .withConfigProperty(XlsInputFormatConfig.NAME_SHEET_VALUE);
      }
```